src/GafferCycles/IECoreCyclesPreview/Renderer.cpp

```diff
@@ -943,6 +944,29 @@ IECore::InternedString g_volumeClippingAttributeName( "cycles:volume_clipping" )
 IECore::InternedString g_volumeStepSizeAttributeName( "cycles:volume_step_size" );
 IECore::InternedString g_volumeObjectSpaceAttributeName( "cycles:volume_object_space" );
 
+const char *customAttributeName( const std::string &attributeName, bool *hasPrecedence = nullptr )
+{
+	if( boost::starts_with( attributeName, "user:" ) )
+	{
+		if( hasPrecedence )
```

Comment:
Since we always want to pass and fill `hasPrecedence`, we might as well pass `bool &` rather than `bool *`, and then we can avoid these checks.

Reply:
Hmm just re-thinking this, the idea is that if there's 2 exactly named attributes eg. `render:myAttr` and `user:myAttr` the `render:` takes precedence over `user:`, but one has `render:` omitted so both could live in the same vector right? Is the idea to just not have the `user:` sent to the renderer regardless of both being able to be available at the same time?

Reply:
`testCustomAttributePrecedence` answered that for me :)